ideascube/serveradmin/catalog.py

```diff
@@ -126,6 +129,10 @@ def remove(self, package):
         self._removed.append(package)
 
     def commit(self):
+        visible_packages = get_config('ideascube', 'displayed-package-ids')
+        [visible_packages.remove(p.id) for p in self._removed if p.id in visible_packages]
```

Comment:
Wow. This is a bit convoluted. You're creating a list which you don't keep, just for the side-effect of modifying another list.

I'd prefer something like this:

```python
for pkg in self._removed:
    try:
        visible_packages.remove(pkg.id)
    except ValueError:
        # It's not in there, nothing to remove them
        pass
```

Or we could make `self._removed` (and `self._installed`) a dictionary, using the package ids as keys, and then you could do:

```python
visible_packages = [id for id in visible_packages if id not in self._removed]
```

Or even something like this, keeping `self._removed` the list it currently is, why not:

```python
removed_ids = [p.id for p in self._removed]
visible_packages = [id for id in visible_packages if id not in removed_ids]
```

Or let's be crazy !

```python
removed_ids = [p.id for p in self._removed]
visible_packages = list(set(visible_packages) - set(removed_ids))
```

Each of these requires less mind contortion than yours, because they are variation of the typical `filter` pattern, and this is exactly what you are trying to do: filtering elements out of a list.

I honestly needed to pause and re-read your line a few times, drawing the connections mentally to figure out what it was doing.

Simple things should be done in obvious ways. Filtering items out of a list is a very simple (and extremely common) thing to do.